### ISSUE TITLE
[BUG] Link from Cursive::new to CursiveExt doesn't work offline.

### ISSUE THREAD

**Issue by masklinn** [gyscos/cursive#584]
I'm not sure how to resolve this. I think the ease of getting and using documentation offline is one of the strength of Rust, but because `cursive_core::Cursive::new` is linking *downstream* it can't use "local" links, even though it's linking to a sibling crate, and will usually be generated as part of the `cursive` documentation & re-export.

Somewhat linked to rust-lang/rust#24305 "Pain points of reexports"
*Definitely* linked to rust-lang/rust#74481 "Figure out a way to link to downstream crates"

**Comment by gyscos** [comment]
Right, that's one of the issues with the current crate organization. As mentioned in the linked threads, one workaround would be to re-document the re-export in the `cursive` crate itself, but that would:
* Prevent using `use cursive_core::*;`, and instead force me to import each module one by one.
* Duplicate the documentation between the two crates, with the risk of one getting outdated or even diverging. _Maybe_ this could be solved by instead doc-linking to a shared markdown file?
* We can only re-document items like types, modules and free functions this way, not methods on a type.

So for now, I think hard-coding the link to docs.rs is the lesser evil.
